homeassistant/helpers/entity.py

```diff
@@ -309,17 +335,29 @@ def device_info(self) -> DeviceInfo | None:
     @property
     def device_class(self) -> str | None:
         """Return the class of this device, from component DEVICE_CLASSES."""
-        return self._attr_device_class
+        if hasattr(self, "_attr_device_class"):
+            return self._attr_device_class
+        if self.entity_description:
+            return self.entity_description.device_class
+        return None
 
     @property
     def unit_of_measurement(self) -> str | None:
         """Return the unit of measurement of this entity, if any."""
-        return self._attr_unit_of_measurement
+        if hasattr(self, "_attr_unit_of_measurement"):
+            return self._attr_unit_of_measurement
+        if self.entity_description:
+            return self.entity_description.unit_of_measurement
+        return None
 
     @property
     def icon(self) -> str | None:
         """Return the icon to use in the frontend, if any."""
-        return self._attr_icon
+        if hasattr(self, "_attr_icon"):
+            return self._attr_icon
+        if self.entity_description:
```

Comment:
Shall we optimize for the description being available case and move this if-statement above the other hasattr check in all methods?

Reply:
nevermind that makes no sense, we should allow entities overriding the description.

Reply:
I think it should be this way around. Mainly that if one overrides or composes the attribute with the description, it still works.

For example:

```py
def __init__(self, device_name, entity_description):
  """Bla bla bla."""
  _attr_name = f"{device_name} {entity_description.name}"
```

So in this case, the attribute should win over the entity description.